RealmSwift/Results.swift

```diff
@@ -164,14 +176,27 @@ extension AnyRealmValue: AddableType {}
      */
     public subscript(position: Int) -> Element {
         throwForNegativeIndex(position)
+        if let projector = projector {
```

Comment:
This seems like something that could just be done in `dynamicBridgeCast` without having the projector at all. `Projection` needs to conform to a protocol that allows constructing it from an ObjectBase, and then `dynamicBridgeCast()` does `if let object = x as? ObjectBase, type = T.self as? ProjectionProtocol.Type { return type(x) as! T }`. This would make it so that none of the getter functions need changes.